tests/driver_atwinc15x0/Kconfig

```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2021 Gunar Schorcht
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config APPLICATION
+    bool
+    default y
+    imply MODULE_ATWINC15X0
+    imply DRIVER_NETDEV_COMMON
```

Comment:
Application configurations of visible symbols are done through `.config` files. During migration we use a special name for testing.

https://github.com/RIOT-OS/RIOT/blob/85313ccc02cdb813d65f208c56d06992ad484130/tests/driver_bmx280/app.config.test#L1-L6

Reply:
I know. In the case of `tests/driver_atwinc15x0`, the `Makefile` is including a common `Makefile.netdev.mk` that is used by all netdev driver tests from `tests/driver_netdev_common` which enables all required modules.

Therefore, I was looking for a possibility to do the same in `Kconfig`. Since I found a `Kconfig` file in a number of tests, e.g. `tests/periph_gpio`, I thought that this could be the possible way because `Kconfig` can source other `Kconfig` files. Using `app.config.test` I have to enable all these modules, that have to enabled by each other netdev driver test application again and again:
```
CONFIG_MODULE_ATWINC15X0=y
CONFIG_MODULE_SHELL=y
CONFIG_MODULE_SHELL_COMMANDS=y
CONFIG_MODULE_PS=y
CONFIG_MODULE_GNRC=y
CONFIG_MODULE_AUTO_INIT_GNRC_NETIF=y
CONFIG_MODULE_GNRC_TXTSND=y
CONFIG_MODULE_GNRC_PKTDUMP=y
```
Furthermore, with these definitions I get errors for all GNRC modules. That's why I was asking in comment https://github.com/RIOT-OS/RIOT/pull/13135#issuecomment-991642190 how to enable GNRC modules in `Kconfig`.
```
[genconfig.py]:ERROR-Treating Kconfig warnings as errors:
[genconfig.py]:ERROR-=> /home/gs/src/RIOT-Xtensa-ESP.working/tests/driver_atwinc15x0/app.config.test:7: warning: attempt to assign the value 'y' to the undefined symbol MODULE_GNRC
[genconfig.py]:ERROR-=> /home/gs/src/RIOT-Xtensa-ESP.working/tests/driver_atwinc15x0/app.config.test:8: warning: attempt to assign the value 'y' to the undefined symbol MODULE_AUTO_INIT_GNRC_NETIF
[genconfig.py]:ERROR-=> /home/gs/src/RIOT-Xtensa-ESP.working/tests/driver_atwinc15x0/app.config.test:9: warning: attempt to assign the value 'y' to the undefined symbol MODULE_GNRC_TXTSND
[genconfig.py]:ERROR-=> /home/gs/src/RIOT-Xtensa-ESP.working/tests/driver_atwinc15x0/app.config.test:10: warning: attempt to assign the value 'y' to the undefined symbol MODULE_GNRC_PKTDUMP
```

Reply:
> Application configurations of visible symbols are done through `.config` files. 

If I understand it right
- `FEATURES_REQUIRED += xyz` is modeled in `app.config.test` as `CONFIG_MODULE_XYZ = y`
- `FEATURES_OPTIONAL += xyz` is modeled as part of `config APPLICATION` in `Kconfig` as `imply
MODULE_XYZ`
- `USEMODULE += xyz` is modeled in `app.config.test` as `CONFIG_MODULE_XYZ = y`

How could it be modeled that `tests/driver_atwinc15x0/Makefile` includes a common `tests/driver_netdev_common/Makefile.netdev.mk` that is used by all netdev driver tests to enable all required modules and to avoid to do it again and again for each netdev driver test? Would it be correct to model it like in e6daec949e70fcfe6dbf86dba8ea0a7971c4a2c9 and e49d21d55dda7f867ba38351091d7befa2adf397?

Reply:
I think the way would be to have a `.config` file and add it to the loaded configurations by adding it to the list of the `KCONFIG_ADD_CONFIG` makefile variable.